__tests__/AppCatalog.js

```diff
@@ -22,162 +26,633 @@ import {
+    getMockCall('/v4/releases/', releasesResponse);
+
+    const clusterDetailPath = RoutePath.createUsablePath(
+      OrganizationsRoutes.Clusters.Detail,
+      {
+        orgId: ORGANIZATION,
+        clusterId: V4_CLUSTER.id,
+      }
+    );
+    const { findByText, getByText } = renderRouteWithStore(clusterDetailPath);
+
+    const appsTab = await findByText(/^apps$/i);
+    fireEvent.click(appsTab);
+
+    // Click on app details button to open the editing modal
+    const appLabel = getByText(/app version: 0.0.1/i);
+    const appDetailsButton = appLabel.parentNode.parentNode.querySelector(
+      'button'
+    );
+    fireEvent.click(appDetailsButton);
+
+    // Upload a configmap file
+    const fileInputPlaceholder = getByText(/configmap has been set/i);
+    let deleteButton = fileInputPlaceholder.parentNode.querySelector(
+      '.btn-danger'
+    );
+    fireEvent.click(deleteButton);
+
+    // Confirm deletion
+    deleteButton = getByText(/^delete configmap$/i);
+    fireEvent.click(deleteButton);
+
+    deleteAppConfigRequest.done();
+  });
+
+  it('updates secrets of an already installed app', async () => {
+    const updateAppConfigRequest = nock(API_ENDPOINT)
+      .intercept(`/v4/clusters/${V4_CLUSTER.id}/apps/my%20app/secret/`, 'PATCH')
+      .reply(StatusCodes.Ok);
+
+    getMockCall(`/v4/organizations/${ORGANIZATION}/credentials/`);
+    getMockCall(`/v4/clusters/${V4_CLUSTER.id}/`, v4AWSClusterResponse);
+    getMockCall(
+      `/v4/clusters/${V4_CLUSTER.id}/status/`,
+      v4AWSClusterStatusResponse
+    );
+    getMockCallTimes(
+      `/v4/clusters/${V4_CLUSTER.id}/apps/`,
+      [appResponseWithCustomConfig],
+      2
+    );
+    getMockCall(`/v4/clusters/${V4_CLUSTER.id}/key-pairs/`);
+    getMockCall('/v4/releases/', releasesResponse);
+
+    const clusterDetailPath = RoutePath.createUsablePath(
+      OrganizationsRoutes.Clusters.Detail,
+      {
+        orgId: ORGANIZATION,
+        clusterId: V4_CLUSTER.id,
+      }
+    );
+    const { findByText, getByText } = renderRouteWithStore(clusterDetailPath);
+
+    const appsTab = await findByText(/^apps$/i);
+    fireEvent.click(appsTab);
+
+    // Click on app details button to open the editing modal
+    const appLabel = getByText(/app version: 0.0.1/i);
+    const appDetailsButton = appLabel.parentNode.parentNode.querySelector(
+      'button'
+    );
+    fireEvent.click(appDetailsButton);
+
+    // Upload a secrets file
+    const fileInputPlaceholder = getByText(/secret has been set/i);
+    const fileInput = fileInputPlaceholder.parentNode.querySelector('input');
+    const file = new Blob(
+      [
+        JSON.stringify({
+          name: 'test',
+          namespace: 'some-other-test',
+        }),
+      ],
+      {
+        type: 'application/json',
+        name: 'secret.json',
+      }
+    );
+    fireEvent.change(fileInput, {
+      target: {
+        files: [file],
+      },
+    });
+
+    await waitForDomChange();
```

Comment:
what are we actually asserting with `waitForDomChange`? Is this enough to say that Happa did what we expected it to do?